### DISCUSSION TITLE
Multiple Environments Deployment Strategy

### DISCUSSION THREAD

**Discussion by gsheppard** [vercel/community#2435] [Help]
### Problem Description

This has been a question I've asked a few sources, but still searching for an answer to.

Vercel is pretty opinionated when it comes to environments, and when you leave the 3 supported environments it gets messy. For reasons out of my control, we have 4 environments: `local/development/staging/production`. The latter 3 are deployable.

To make this work we have 3 environment-based branches for the deployable environments (`development/staging/production`), and we've been forced to adopt a git flow deployment strategy, _which I hate_. I'd love to have multiple environments assigned to one branch (e.g. `main` gets deployed to both `development` and `staging` environments) and then just simply "Promote to production" when necessary, but I don't think that is even possible since the environment variables will be the same between the two.

**Some questions:**
- Does anyone else deal with multiple environments that Vercel doesn't agree with?
- What has been your strategy?
- Are there tricks to the CLI I can leverage with Github Actions that would eliminate the need for so many primary branches?

**Comment by MatteoGauthier** [comment]
I also find it complicated to manage several environments with Vercel. I think users should be able to create their own deployment environments. I'm moving from multiple environments (staging, recette, prod, preprod), managing credentials with multiple Preview branch envs is really complicated
